### PR TITLE
Notebook improvements

### PR DESCRIPTION
**Pull request by philippjfr**
Various notebook improvements:

- [x] Added list of supported data formats to Tabular Datasets user guide
- [x] Fixed Plotting with Bokeh axis example
- [x] Added section about bokeh interactive legends to Plotting with Bokeh  (as requested in https://github.com/ioam/holoviews/issues/1265)

### CODE REVIEW

File: examples/user_guide/07-Tabular_Datasets.ipynb [hunk 1]
```diff
@@ -84,12 +104,10 @@
...
-    "* 0D: BoxWhisker, Spikes, Distribution*, \n",
-    "* 1D: Scatter, Curve, ErrorBars, Spread, Bars, BoxWhisker, Regression*\n",
-    "* 2D: Points, HeatMap, Bars, BoxWhisker, Bivariate*\n",
-    "* 3D: Scatter3D, TriSurface, VectorField, BoxWhisker, Bars\n",
-    "\n",
-    "\\* - requires Seaborn\n",
+    "* 0D: BoxWhisker, Spikes, Distribution, \n",
+    "* 1D: Scatter, Curve, ErrorBars, Spread, Bars, BoxWhisker\n",
+    "* 2D: Points, HeatMap, Bars, BoxWhisker, Bivariate\n",
+    "* 3D: Scatter3D, Trisurface, VectorField, BoxWhisker, Bars\n",
```
Comment (jlstevens) [inline examples/user_guide/07-Tabular_Datasets.ipynb]: Should now be ``TriSurface``.

**Comment by jlstevens** [comment]
Looks good, other than updating the spelling to ``TriSurface``, I am happy to merge.

File: examples/user_guide/Plotting_with_Bokeh.ipynb [hunk 2]
```diff
@@ -244,15 +244,15 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%%opts Overlay [show_legend=False height=400 width=600] ErrorBars (line_width=5) Scatter(alpha=0.2 size=6)\n",
+    "%%opts Overlay [show_legend=False height=400 width=600] ErrorBars (line_width=5) Scatter [jitter=0.2] (alpha=0.2 size=6)\n",
```
Comment (jlstevens) [inline examples/user_guide/Plotting_with_Bokeh.ipynb]: Just checking that ``jitter`` complains if it is used without categorical axes (it should)....

File: examples/user_guide/07-Tabular_Datasets.ipynb (hunk 1 above)
Comment (philippjfr) [reply examples/user_guide/07-Tabular_Datasets.ipynb]: Good catch.

File: examples/user_guide/Plotting_with_Bokeh.ipynb (hunk 2 above)
Comment (philippjfr) [reply examples/user_guide/Plotting_with_Bokeh.ipynb]: Opened an issue here: https://github.com/ioam/holoviews/issues/2236

**Comment by philippjfr** [comment]
@jlstevens Ready to merge.

**Comment by jlstevens** [comment]
Looks good. Merging!

**Pull request merged**